src/main/java/duke/command/Parser.java

```diff
@@ -0,0 +1,188 @@
+package duke.command;
+
+import java.time.LocalDate;
+import java.time.format.DateTimeFormatter;
+import java.time.format.DateTimeParseException;
+
+import java.util.regex.Matcher;
+import java.util.regex.Pattern;
+
+import duke.task.Deadline;
+import duke.task.Event;
+import duke.task.TaskList;
+import duke.task.Todo;
+
+/**
+ * Parser class which is used to process commands passed down by the user and returns
+ * the result of each command as a String.
+ */
+public class Parser {
+
+    private TaskList list;
+    private boolean isExit;
+
+    public Parser(TaskList list) {
+        this.list = list;
+        this.isExit = false;
+    }
+
+    /**
+     * Checks if the command to exit has been issued
+     * @return True if 'bye' command has been issued, False otherwise.
+     */
+    public boolean isExit() {
+        return isExit;
+    }
+
+    /**
+     * The function that processes the incoming command and executes different
+     * functions depending on the command.
+     * @param command String command provided by the user.
+     * @return String result of the provided command.
+     * @throws DukeException A custom Exception that carries a message for the user if thrown.
+     */
+    public String processCommand(String command) throws DukeException {
+        Pattern pattern = Pattern.compile("^(.*?)\\s(.*?)(?:\\s/..\\s?(.*))?$");
+        Matcher matcher = pattern.matcher(command);
+        if (command.equals("list")) {
+            return this.list.toString();
+        }
+        if (command.equals("hello")) {
+            return "Hi! I'm Duke! Pleasure to meet you :)";
+        }
+        if (command.equals("bye")) {
+            this.isExit = true;
+            return "Bye! Hope to see you again soon!";
+        }
+        if (command.equals("help")) {
+            return "Accepted commands:\n" +
+                    "hello - hello!\n" +
+                    "list - show current list\n" +
+                    "bye - saves the current list and exits the program\n" +
+                    "\n" +
+                    "todo <description> - create a todo Task\n" +
+                    "event <description> /at <dd/MM/yyyy> - create an event Task (date is optional)\n" +
+                    "deadline <description> /by <dd/MM/yyyy> - create a deadline Task (date is optional)\n" +
+                    "\n" +
+                    "done <index> - mark the specified task as done\n" +
+                    "undo <index> - mark the specified task as not done\n" +
+                    "delete <index> - deletes the specified task from the list";
+        }
+        if (matcher.find()) {
+            String com = matcher.group(1);
```

Comment:
Will it be better to give a more descriptive name and include the information on which part of the command it represents?